libcontainer/process_linux.go

```diff
@@ -89,27 +96,35 @@ func newProcessComm() (*processComm, error) {
 
 func (c *processComm) closeChild() {
 	_ = c.initSockChild.Close()
+	_ = c.stage1SockChild.Close()
 	_ = c.syncSockChild.Close()
 	_ = c.logPipeChild.Close()
 }
 
 func (c *processComm) closeParent() {
 	_ = c.initSockParent.Close()
+	_ = c.stage1SockParent.Close()
 	_ = c.syncSockParent.Close()
 	// c.logPipeParent is kept alive for ForwardLogs
 }
 
+type containerProcess struct {
+	cmd           *exec.Cmd
+	comm          *processComm
+	config        *initConfig
+	manager       cgroups.Manager
+	fds           []string
+	process       *Process
+	bootstrapData io.Reader
+	container     *Container
+	childPid      int
+}
+
 type setnsProcess struct {
-	cmd             *exec.Cmd
-	comm            *processComm
+	containerProcess
 	cgroupPaths     map[string]string
 	rootlessCgroups bool
-	manager         cgroups.Manager
 	intelRdtPath    string
-	config          *initConfig
-	fds             []string
-	process         *Process
-	bootstrapData   io.Reader
```

Comment:
Can we have this as a separate PR?

It's a nice refactor (wanted to do something like this but never got around to) but it would be easy to review it separately.

Reply:
I'll do it later, I should let all CI tests passed first.